### ISSUE TITLE
OpenGL binding missing byte[] overloads for functions with data pointer arguments.

### ISSUE THREAD

**Issue by theagentd** [LWJGL/lwjgl3#547]
* LWJGL version: 3.2.3
* Java version: 13
* Platform: All
* Module: OpenGL

The OpenGL binding is missing byte[] overloads for all (?) functions that take in a data pointer:
- glBufferData()
- glBufferSubData()
- glBufferStorage()
- glNamedBufferData()
- glNamedBufferSubData() <-- specifically what I needed
- glNamedBufferStorage()
- glClear*()
- gl*Tex*Image()

All these functions support primitive arrays such as float[] and int[] but not byte[].

**Comment by Spasi** [comment]
This is by design, explained [here](https://github.com/LWJGL/lwjgl3/issues/193#issuecomment-223880662).

**Comment by theagentd** [comment]
Thank you for the response!

Considering how many overloaded versions of functions for uploading data to OpenGL we already have, adding support for the single last primitive array in Java doesn't really feel like it'd make a big difference bloat-wise. :) 

In our case, we read in serialized data from disk and end up with some metadata and a byte[]. Due to the libraries/existing code we have, we can't read the data directly into a direct ByteBuffer. In our case, it would be very useful to have byte[] overloads so we don't need to copy the data to a direct ByteBuffer just to upload it to a buffer/texture.

It is also not that unusual to process byte arrays directly in the case of image processing, just saying.

In short, my main argument for byte[] overloads is in the case where your data for some reason is sourced from a byte[] and you are not able to change that code.

**Comment by XenoAmess** [comment]
no you still have to.
byte[] is in jvm heap and can never be handled by native code directly.

XenoAmess

________________________________
